compiler/optimizations/gpuTransforms.cpp

```diff
@@ -46,6 +46,23 @@ static bool debugPrintGPUChecks = false;
 static bool allowFnCallsFromGPU = true;
 static int indentGPUChecksLevel = 0;
 
+// Ideally, if we do gpuSpecialization, we could safely assume that any function
+// that isn't marked with FLAG_GPU_SPECIALIZE would be executed on a CPU locale.
+// Unfortunately, as of today, this isn't the case because these functions can
+// be reached by virtual dispatch and our specialization cloning isn't
+// sophisticated enough to clone these functions and update the dispatch calls
+// as appropriate.
+//
+// The good news is it's still safe to assume that anything marked with
+// FLAG_GPU_SPECIALIZE must execute on the gpu.
+//
+// Anyway, of course, our hope would be to make things handle virtual dispatch
+// properly, but, in the meantime, if you want to ignore this and make
+// these assumptions anyway you can turn this flag to true. And once we do
+// update things we can get just simplify things and get rid of this flag
+// altogether.
+static bool recklesslyDisregardSpecializationLimits = false;
```

Comment:
not to bikeshed, but "limits" part is not helpful to me and "virtual" in the name could be a bit better. "disregard possible virtual calls" and true-by-def maybe?

You can take my suggestion or something similar if you like it, I don't intend to argue too much about the name.

Reply:
How about: `assumeNonGpuSpecFnsAreOnCpu`.

That doesn't really capture when its safe for this to be true, but I think my hope is the name is ugly enough people look at the comment at its definition to figure out "why" this exists.

Reply:
And "false" is the value, right? For some reason I can't think about `false` by default here. It's neither here nor there, you can ignore my rambling. It took me couple of seconds thinking "isn't that the opposite of what we're doing". But yeah, I'm OK with that name.

Reply:
> And "false" is the value, right?

Correct, I'm thinking of "false" with synonymous with "no" here.

So with this PR I'm no longer (incorrectly) assuming that non-gpu-specialized functions are only run on a CPU.